### PR TITLE
Switch twitter to WS

### PR DESCRIPTION
**Pull request by brian-brazil**
A few small improvements in passing too. I kept what old code I could, rather than rewriting everything.

The defaults here could also do with work, it should be tweeting scores at every jam - not every ~15 minutes.

### CODE REVIEW

File: html/controls/twitter.js [hunk 1]
```diff
@@ -1,165 +1,185 @@
...
-				success: function(desc) {
-					keys.push({ label: e+" ("+desc+")", value: e });
-				}
-			});
-		});
-		setupInputFieldAutoComplete(keys);
-	});
+$(function() {
+  WS.AutoRegister();
+  WS.Connect();
```
Comment (frank-weinberg) [inline html/controls/twitter.js]: Don't you usually put these at the bottom for performance?

File: src/com/carolinarollergirls/scoreboard/core/impl/TwitterImpl.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+import twitter4j.Status;
+import twitter4j.TwitterException;
+import twitter4j.AsyncTwitter;
+import twitter4j.AsyncTwitterFactory;
+import twitter4j.TwitterAdapter;
+import twitter4j.TwitterListener;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/TwitterImpl.java]: unused

File: src/com/carolinarollergirls/scoreboard/core/impl/TwitterImpl.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+import twitter4j.TwitterMethod;
+import twitter4j.auth.AccessToken;
+import twitter4j.auth.RequestToken;
+import twitter4j.conf.ConfigurationBuilder;
+
+import com.carolinarollergirls.scoreboard.core.Clock;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/TwitterImpl.java]: unused

File: src/com/carolinarollergirls/scoreboard/core/impl/TwitterImpl.java [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+                }
+            }
+                                             ));
+        }
+        addWriteProtection(Child.FORMAT_SPECIFIER);
+
+        twitter.addListener(new Listener());
+    }
+
+    public void postAutosaveUpdate() {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/TwitterImpl.java:66]: This should have an @Override annotation

File: src/com/carolinarollergirls/scoreboard/core/Twitter.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+/**
+ * Copyright (C) 2008-2012 Mr Temper <MrTemper@CarolinaRollergirls.com>
+ *
+ * This file is part of the Carolina Rollergirls (CRG) ScoreBoard.
+ * The CRG ScoreBoard is licensed under either the GNU General Public
+ * License version 3 (or later), or the Apache License 2.0, at your option.
+ * See the file COPYING for details.
+ */
+
+import java.util.Map;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/Twitter.java]: unused

File: src/com/carolinarollergirls/scoreboard/core/Twitter.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.AddRemoveProperty;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.CommandProperty;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.PermanentProperty;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEvent.ValueWithId;
+import com.carolinarollergirls.scoreboard.event.ScoreBoardEventProvider;
+import com.carolinarollergirls.scoreboard.utils.ValWithId;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/Twitter.java]: unused

File: src/com/carolinarollergirls/scoreboard/core/impl/ScoreBoardImpl.java [hunk 7]
```diff
@@ -46,6 +47,9 @@ public ScoreBoardImpl() {
     }
 
     protected void setupScoreBoard() {
+        if (getCurrentTimeout() == null) {
+          set(Value.CURRENT_TIMEOUT, noTimeoutDummy);
+        }
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/ScoreBoardImpl.java]: What is this doing here?

**Review by frank-weinberg — COMMENTED** [review]
Having never even looked into the Twitter functionality, I have not tested any of this.

File: src/com/carolinarollergirls/scoreboard/core/impl/ScoreBoardImpl.java (hunk 7 above)
Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/core/impl/ScoreBoardImpl.java]: Preventing an NPE, but I handled it in another way.

**Review by brian-brazil — COMMENTED** [review]
All addressed.

File: src/com/carolinarollergirls/scoreboard/core/Twitter.java (hunk 6 above)
Comment (frank-weinberg) [reply src/com/carolinarollergirls/scoreboard/core/Twitter.java]: This one is still there.

Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/core/Twitter.java]: Fixed

**Pull request merged**